crates/network-scheduler/src/storage.rs

```diff
@@ -301,6 +308,53 @@ impl S3Storage {
         prometheus_metrics::s3_request();
     }
 
+    pub async fn save_assignment(&self, scheduler_state: &Vec<u8>) {
+        log::debug!("Encoding assignment");
+        let json: Value = serde_json::from_slice(scheduler_state).unwrap();
+        let assignment = Assignment::new(&json, Config::get().cloudflare_storage_secret.clone());
+        let mut encoder = GzEncoder::new(Vec::new(), Compression::best());
+        let _ = encoder.write_all(serde_json::to_vec(&assignment).unwrap().as_slice());
+        let compressed_bytes = encoder.finish().unwrap();
+        log::debug!("Saving assignment");
+        let mut hasher = Sha256::new();
+        hasher.update(compressed_bytes.as_slice());
+        let hash = hasher.finalize();
+        let network = Config::get().network.clone();
+        let dt = Utc::now();
+        let ts = dt.format("%Y%m%dT%H%M%S");
+        let filename: String =  format!("assignments/{network}_{ts}_{hash:X}.json.gz");
+        
+        let _ = self
+            .client
+            .put_object()
+            .bucket(&self.config.scheduler_state_bucket)
+            .key(&filename)
+            .body(compressed_bytes.into())
+            .send()
+            .await
+            .map_err(|e| log::error!("Error saving assignment: {e:?}"));
```

Comment:
If an error occurs here, we should return from this function. It's easier if it returns a `Result` and the logging is done outside of it

Reply:
fixed